app/scripts/controllers/transactions/tx-state-manager.test.js

```diff
@@ -614,6 +631,136 @@ describe('TransactionStateManager', function () {
       );
       assert.equal(result[1].id, 2, 'early txs truncated');
     });
+
+    it('cuts off entire groups of transactions by nonce when adding new transaction', function () {
+      const limit = txStateManager.txHistoryLimit;
+      // In this test case the earliest two transactions are a dropped attempted ether send and a
+      // following cancel transaction with the same nonce. these two transactions should be dropped
+      // together as soon as the 11th unique nonce is attempted to be added, mocked here with limit + 1
+      const txs = generateTransactions(limit + 2, {
```

Comment:
Shouldn't this be `limit + 1`? :thinking: That's what the description implies. And if I'm understanding this correctly, `limit + 1` is when the behavior we're testing is executed.

Reply:
the text is confusing and probably needs updated, but limit applies to unique nonces now, and so one of the + 1s is to get us into the state where we have a group of transactions at the beginning of the array that are together, the second + 1 is to push us over the limit of unique nonces.

Reply:
at the very least the limit + 1 should be removed :P

Reply:
Ah right, gotcha. Makes sense.

Reply:
modifed the explainer text